review typos + format (#8)
marigold-dev/training-nft-1: README.md

Training n°1 for NFT marketplace

> # Introduction

Author: I feel like this introduction is not an introduction.
Your real introduction is in the next part:
> We are going to build a wine marketplace

I think it should be `preface` or `foreword` (don't know which one is the good translation of `avant-propos`).

Business objects managed by a blockchain are called `assets`. On Tezos you will find the term `Financial Asset or FA` with different version 1, 2 or 2.1.
Here below different categorization of assets.

![](http://jingculturecommerce.com/wp-content/uploads/2021/03/nft-assets-1024x614.jpg)

# :wine_glass: Wine marketplace

> We are going to build a Wine marketplace.

Author: Please add details!
What features will be available on this marketplace?

A quick list like:

> You are going to build a wine marketplace:
>- add new wine bottles
>- manage your personal wine collection
>- sell some of your bottles
>- buy bottle(s) of other people

Author: You can also say that we will focus on the contract code, and so, you will provide the related front-end code to copy/paste.

You can play with the [final demo](https://demo.winefactory.marigold.dev/).

![nftfactory.png](./doc/nftfactory.png)

Plan of the training course :

- [NFT 1](https://github.com/marigold-dev/training-nft-1): use FA2 NFT template to understand the basics

> - [NFT 2](https://github.com/marigold-dev/training-nft-2): finish FA2 nft marketplace

Author: ```suggestion
- [NFT 2](https://github.com/marigold-dev/training-nft-2): finish FA2 NFT marketplace
```

## :scroll: Smart Contracts

We will use two contracts for the marketplace.

### The token contract.

> On Tezos FA2 is the standard for Non-fungible Token contracts. We will be using the [template provided by Ligo](https://packages.ligolang.org/package/@ligo/fa) to build out the Token Contract. The template contains the basic entrypoints for building a Fungible or Non-fungible token including:

Author: ```suggestion
FA2 is the standard for Non-fungible Token contracts on Tezos. We will be using the [template provided by Ligo](https://packages.ligolang.org/package/@ligo/fa) to build out the Token Contract. The template contains the basic entrypoints for building a Fungible or Non-fungible token including:
```

     });
```

Explanations:

> - the first line `#import "@ligo/fa/lib/fa2/nft/NFT.jsligo" "NFT"` imports the ligo FA library

Author: ![image](https://user-images.githubusercontent.com/45454460/211520780-62e218a7-433e-4e0f-99bd-1c1c6f8d6705.png)

This is obvious, anybody can understand we import it. You should explain **why** we import it.

> - `storage` definition is an extension of the imported library storage `(NFT.Ledger.t,NFT.Metadata.t,NFT.TokenMetadata.t,NFT.Operators.t,set<NFT.Storage.token_id>)`

Author: Need to be rephrased. Maybe to something like:
> - `storage` type is the storage imported from library: `(NFT.Ledger.t,NFT.Metadata.t,NFT.TokenMetadata.t,NFT.Operators.t,set<NFT.Storage.token_id>)` but with this and that because of this and that.
```

Author: Please also rework parameter. Because
> `parameter` has an entrypoint `AddAdministrator`to add new administrators
is kinda useless... Either your explain what is an administrator, and why we need them for the training, or you don't mention them in the explanations.

      }
```

Explanations:

- the called function is taking the storage type of the library, so we send a partial object from our storage definition to match the type definition

> - the return type contains also the storage type of the library, so we need to reconstruct the storage by copied the modified fields

Author: `the called function` is `NFT.transfer`? If yes, please use its name.

     });
```

Explanations:

- `mint` function will allow you to create a unique NFT. You have to declare the name, description, symbol and ipfsUrl for the picture to display

> - to simplify, we don't manage the increment of the token_id here it will be done by the front end later. We encourage you to manage this counter onchain to avoid overriding and existing nft. There is no rule to allocate a specific number to the token_id but people increment it from 0. Also, there is no rule if you have a burn function to reallocate the token_id to a removed index and just continue the sequence from the greatest index.

Author: ```suggestion
- to simplify, we don't manage the increment of the `token_id` here it will be done by the front end later. We encourage you to manage this counter on chain to avoid overriding and existing NFT. There is no rule to allocate a specific number to the `token_id` but people increment it from 0. Also, there is no rule if you have a burn function to reallocate the `token_id` to a removed index and just continue the sequence from the greatest index.
```

└──────────┴──────────────────────────────────────┴───────┴──────────────────┴────────────────────────────────┘
```

**:tada: Hooray! We have finished the backend :tada:**

# :performing_arts: NFT Marketplace front

## Get the react boilerplate

To win time, we have a [boilerplate ready for the UI](https://github.com/marigold-dev/training-nft-1/tree/main/reactboilerplateapp)

Copy this code into your folder (:warning: assuming you have cloned this repo, and your current path is `$REPO/training`)

```bash
cp -r ../reactboilerplateapp/ ./app
```

> Note : if you want to understand how it has been made from scratch look at [this training](https://github.com/marigold-dev/training-dapp-1#construction_worker-dapp)

> Generate the Typescript classes from Michelson code to your frontend app and run the server

Author: Complicated sentence. Can you please reformulate? Maybe splitting it into two different sentences could help.

```bash

> taq install @taqueria/plugin-contract-types

Author: Please briefly describe what each line is doing. I don't copy/paste/execute command if I don't know what they are supposed to do.
